hydra/core/utils.py

```diff
@@ -166,14 +174,32 @@ def register(name: str, f: Any) -> None:
     register("python_version", lambda level="minor": version_dict.get(level))
 
 
+class JobResult(Enum):
+    COMPLETED = 1
+    FAILED = 2
+
+
 @dataclass
 class JobReturn:
     overrides: Optional[Sequence[str]] = None
-    return_value: Any = None
     cfg: Optional[DictConfig] = None
     hydra_cfg: Optional[DictConfig] = None
     working_dir: Optional[str] = None
     task_name: Optional[str] = None
+    job_result: Optional[JobResult] = None
+    _return_value: Any = None
+
+    @property
+    def return_value(self) -> Any:
+        assert self.job_result is not None, "job_result not yet available"
+        if self.job_result == JobResult.COMPLETED:
+            return self._return_value
+        else:
+            raise HydraJobException("Job failed.") from self._return_value
```

Comment:
I feel "Job failed." is not super helpful.
How about:
```suggestion
            raise HydraJobException("Error executing job : {self.overrides}") from self._return_value
```

Reply:
What does it look like if you use the regular HydraException instead of your special one?